File: 0260_single_number_iii/single_number.go
package main

func singleNumber(nums []int) []int {
	xor := 0
	for _, v := range nums {
		xor = xor ^ v
	}
	lastBit := xor & (-xor)
	res := make([]int, 2)
	for _, v := range nums {
		if v&lastBit == 0 {
			res[0] ^= v
		} else {
			res[1] ^= v
		}
	}
	return res
}

func main() {

}

File: 0260_single_number_iii/single_number.py
from typing import List


def single_number(nums: List[int]) -> List[int]:
    bitmask = 0
    for num in nums:
        bitmask ^= num
    diff = bitmask & (-bitmask)
    x = 0
    for num in nums:
        if num & diff:
            x ^= num
    return [x, bitmask ^ x]
